src/ring_theory/ideal/basic.lean

```diff
@@ -256,6 +256,31 @@ begin
   rwa ← span_eq I,
 end
 
+lemma has_left_inv_iff_span_top {x : α} :
+  has_left_inv x ↔ span ({x} : set α) = ⊤ :=
```

Comment:
Can you flip the direction of this lemma to match `span_singleton_eq_top` below? So something like
```suggestion
/-- A version of `ideal.span_singleton_eq_top` for non-commutative rings. -/
lemma span_singleton_eq_top' {x : α} :
  span ({x} : set α) = ⊤ ↔ has_left_inv x :=
```

Reply:
OK I changed as you suggested.